src/Dfe.PlanTech.Application/Submission/Commands/SubmitAnswerCommand.cs

```diff
@@ -69,11 +72,32 @@ public async Task<bool> NextQuestionIsAnswered(int submissionId, string nextQues
             return false;
         }
 
+        public async Task<Domain.Questionnaire.Models.Question?> GetNextUnansweredQuestion(int submissionId)
+        {
+            List<QuestionWithAnswer> questionWithAnswerList = await _getLatestResponseListForSubmissionQuery.GetResponseListByDateCreated(submissionId);
+
+            QuestionWithAnswer latestQuestionWithAnswer = questionWithAnswerList[0];
+
+            var question = await _getSubmitAnswerQueries.GetQuestionnaireQuestion(latestQuestionWithAnswer.QuestionRef, null, CancellationToken.None);
+
+            var nextQuestion = Array.Find(question.Answers, answer => answer.Sys.Id.Equals(latestQuestionWithAnswer.AnswerRef))?.NextQuestion;
```

Comment:
What happens if question is null?  What happens in Array.find?

Reply:
Question being null is handled in the called method already (`GetQuestionnaireQuestion()`), throws a KeyNotFoundException.

Array.Find finds the next question from the last answer the user chose before they left the journey, if it's not null then the user is directed to that question as it's where they left off. If its null then it means they were on the check answers page when they left.